flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateDownloader.java

```diff
@@ -120,14 +121,14 @@ private void downloadDataForStateHandle(
 		CloseableRegistry closeableRegistry) throws IOException {
 
 		FSDataInputStream inputStream = null;
-		FSDataOutputStream outputStream = null;
+		OutputStream outputStream = null;
 
 		try {
-			FileSystem restoreFileSystem = restoreFilePath.getFileSystem();
 			inputStream = remoteFileHandle.openInputStream();
 			closeableRegistry.registerCloseable(inputStream);
 
-			outputStream = restoreFileSystem.create(restoreFilePath, FileSystem.WriteMode.OVERWRITE);
+			Files.createDirectories(restoreFilePath.getParent());
+			outputStream = Files.newOutputStream(restoreFilePath, StandardOpenOption.CREATE, StandardOpenOption.WRITE);
```

Comment:
Do we need to add an Option `TRUNCATE_EXISTING` here?

Reply:
I think you are right. I will follow Yu's suggestion below to not supply any option, which implies a `TRUNCATE_EXISTING` option.